### PR TITLE
association tests

### PR DESCRIPTION
**Pull request by kxmbrian**
- Add shoulda gem
- Write association tests
- Fix bugs that were found

### CODE REVIEW

**Comment by lowjoel** [comment]
Please fix the hound violations and squash the commits.

Are you dependent on another PR?

Also, please revert the line changes for Gemfile.lock wrt wdm and tzinfo-data. I'm still wondering if we should keep them, or not. Best not to flip-flop for now.

**Comment by kxmbrian** [comment]
@lowjoel This is ready to be checked/merged

File: spec/models/course/announcement_spec.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 require 'rails_helper'
 
 RSpec.describe Course::Announcement, type: :model do
+  it { should belong_to(:creator).class_name(User.name) }
+  it { should belong_to(:course).inverse_of(:announcements) }
```
Comment (wangqiang1208) [inline spec/models/course/announcement_spec.rb]: @kxmbrian, I think here need to change all the should syntax to expect http://betterspecs.org/#expect

<code> it { is_expected.to XXX } </code>

**Comment by kxmbrian** [comment]
OK, fixed.

**Comment by lowjoel** [comment]
Hound's complaints still seem to be there.

**Pull request merged**